freqtrade/data/history/idatahandler.py

```diff
@@ -147,19 +147,16 @@ def ohlcv_load(self, pair, timeframe: str,
 
         pairdf = self._ohlcv_load(pair, timeframe,
                                   timerange=timerange_startup)
-        if pairdf.empty:
-            if warn_no_data:
-                logger.warning(
-                    f'No history data for pair: "{pair}", timeframe: {timeframe}. '
-                    'Use `freqtrade download-data` to download the data'
-                )
+        if self._check_empty_df(pairdf, pair, timeframe, warn_no_data):
             return pairdf
         else:
             enddate = pairdf.iloc[-1]['date']
 
             if timerange_startup:
                 self._validate_pairdata(pair, pairdf, timerange_startup)
                 pairdf = trim_dataframe(pairdf, timerange_startup)
+                if self._check_empty_df(pairdf, pair, timeframe, warn_no_data):
+                    return pairdf
 
             # incomplete candles should only be dropped if we didn't trim the end beforehand.
             return clean_ohlcv_dataframe(pairdf, timeframe,
```

Comment:
Should also
```
                if self._check_empty_df(pairdf, pair, timeframe, warn_no_data):
                    return pairdf
```
be added here after the call to clean_ohlcv_dataframe? (as I mentioned above, this is simple and fits well this PR)

Actually (and this is not for this PR), we can later extend `_check_empty_df` to log a warning if the length of the df is less than the strategy startup period, warning the user about possible wrong signals in this case.

Reply:
i'm not sure - under what (realistic) circumstance would a non-empty DataFrame come out empty from clean_ohlcv?

Reply:
when it drops single candle

Reply:
that's not very realistic ... you'd have to set the timerange exactly to one candle - which will then remove that ...

Reply:
but why should not we print this message if we print it in other cases when it gets empty?